src/classes/Shapes/BoxShape.js

```diff
@@ -96,33 +107,33 @@ Goblin.BoxShape.prototype.findSupportPoint = function( direction, support_point
  * @property end {vec3} end point of the segment
  * @return {RayIntersection|null} if the segment intersects, a RayIntersection is returned, else `null`
  */
-Goblin.BoxShape.prototype.rayIntersect = (function(){
-	var direction = new Goblin.Vector3(),
-		tmin, tmax,
-		axis, ood, t1, t2, extent;
-
-	return function( start, end ) {
-		tmin = 0;
-
-		direction.subtractVectors( end, start );
-		tmax = direction.length();
-		direction.scale( 1 / tmax ); // normalize direction
-
-		for ( var i = 0; i < 3; i++ ) {
-			axis = i === 0 ? 'x' : ( i === 1 ? 'y' : 'z' );
-			extent = ( i === 0 ? this.half_width : (  i === 1 ? this.half_height : this.half_depth ) );
-
-			if ( Math.abs( direction[axis] ) < Goblin.EPSILON ) {
-				// Ray is parallel to axis
-				if ( start[axis] < -extent || start[axis] > extent ) {
-					return null;
-				}
-			}
-
-            ood = 1 / direction[axis];
-            t1 = ( -extent - start[axis] ) * ood;
-            t2 = ( extent - start[axis] ) * ood;
-            if ( t1 > t2  ) {
+Goblin.BoxShape.prototype.rayIntersect = ( function() {
+    var direction = new Goblin.Vector3(),
+        tmin, tmax,
+        axis, ood, t1, t2, extent;
+
+    return function( start, end ) {
+        tmin = 0;
+
+        direction.subtractVectors( end, start );
+        tmax = direction.length();
+        direction.scale( 1 / tmax ); // normalize direction
+
+        for ( var i = 0; i < 3; i++ ) {
```

Comment:
I know it's painful, but let's unroll this loop: too much branching going on for 3 iterations :)

Reply:
Done. I was able to partially vectorize the implementation as well